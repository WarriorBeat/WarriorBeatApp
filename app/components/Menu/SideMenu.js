/**
 * SideMenu.js
 * SideMenu Component
 * Main File
 */

import React from 'react';
import { View, Image } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { Button } from 'react-native-elements';
import { side_menu } from './styles';

export const MenuButton = props => {
  let btnTextStyle =
    props.textStyle || props.isFooter
      ? side_menu.buttonFooterText
      : side_menu.buttonText;

  let btnStyle =
    props.buttonStyle || props.isFooter
      ? side_menu.buttonFooter
      : side_menu.button;

  return (
    <Button
      large
      backgroundColor={'transparent'}
      textStyle={btnTextStyle}
      buttonStyle={btnStyle}
      title={props.title}
      icon={props.icon}
      onPress={props.onPress}
    />
  );
};

class SideMenu extends React.Component {
  render() {
    return (
      <LinearGradient
        style={side_menu.container}
        colors={['#2d2d2d', '#393939']}
        locations={[0.2, 0.3]}
        start={{ x: 0.5, y: 1.0 }}
        end={{ x: 1, y: 0.25 }}
      >
        <View style={side_menu.container}>
          {this.props.headerImage ? (
            <Image
              source={this.props.headerImage}
              style={side_menu.image}
              resizeMode="cover"
            />
          ) : (
            undefined
          )}
          <View style={side_menu.container_content}>{this.props.children}</View>
          <View style={side_menu.footer_container}>{this.props.footer}</View>
        </View>
      </LinearGradient>
    );
  }
}

export default SideMenu;
