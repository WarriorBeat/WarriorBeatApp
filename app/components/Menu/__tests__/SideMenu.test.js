/**
 * __tests__/SideMenu.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { render, fireEvent } from "react-native-testing-library"
import Providers, { Data, store } from "tests"
import { Text } from "react-native"
import SideMenu from "../SideMenu"

test("should render correctly", () => {
  const tree = render(
    <SideMenu>
      <Text>Hi</Text>
    </SideMenu>,
  )
  expect(tree).toMatchSnapshot()
})

test("should render with logo", () => {
  const tree = render(
    <SideMenu logo>
      <Text>Hi</Text>
    </SideMenu>,
  )
  expect(tree).toMatchSnapshot()
})

test("should render with header, footer, logo", () => {
  const tree = render(
    <SideMenu logo header={<Text>Hi</Text>} footer={<Text>Hi</Text>}>
      <Text>Hi</Text>
    </SideMenu>,
  )
  expect(tree).toMatchSnapshot()
})
