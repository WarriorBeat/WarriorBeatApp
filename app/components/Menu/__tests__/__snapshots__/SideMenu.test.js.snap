// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4281150765,
      4281940281,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.25,
    }
  }
  locations={
    Array [
      0.2,
      0.3,
    ]
  }
  startPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  style={
    Object {
      "color": "transparent",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      center={null}
      hMargin="0%"
      hPad="0%"
      height="100%"
      justify="center"
      size={null}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "height": 1334,
          "justifyContent": "center",
          "marginHorizontal": 0,
          "marginVertical": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": undefined,
        }
      }
      vMargin="0%"
      vPad="0%"
      width={null}
    >
      <View
        center={null}
        hMargin="0%"
        hPad="2.5%"
        height={null}
        justify="flex-start"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 19,
            "paddingVertical": 73.5,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="5.5%"
        width="100%"
      />
      <View
        center={null}
        hMargin="0%"
        hPad="3%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 22.5,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        >
          <Text>
            Hi
          </Text>
        </View>
      </View>
      <View
        center="end"
        hMargin="0%"
        hPad="0%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        />
      </View>
    </View>
  </View>
</BVLinearGradient>
`;

exports[`should render with header, footer, logo 1`] = `
<BVLinearGradient
  colors={
    Array [
      4281150765,
      4281940281,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.25,
    }
  }
  locations={
    Array [
      0.2,
      0.3,
    ]
  }
  startPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  style={
    Object {
      "color": "transparent",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      center={null}
      hMargin="0%"
      hPad="0%"
      height="100%"
      justify="center"
      size={null}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "height": 1334,
          "justifyContent": "center",
          "marginHorizontal": 0,
          "marginVertical": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": undefined,
        }
      }
      vMargin="0%"
      vPad="0%"
      width={null}
    >
      <View
        center={null}
        hMargin="0%"
        hPad="2.5%"
        height={null}
        justify="flex-start"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 19,
            "paddingVertical": 73.5,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="5.5%"
        width="100%"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 150,
                "height": 300,
                "width": 300,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/brand/main.jpg",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        center={null}
        hMargin="0%"
        hPad="3%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 22.5,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        >
          <Text>
            Hi
          </Text>
        </View>
      </View>
      <View
        center="end"
        hMargin="0%"
        hPad="0%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        >
          <Text>
            Hi
          </Text>
        </View>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;

exports[`should render with logo 1`] = `
<BVLinearGradient
  colors={
    Array [
      4281150765,
      4281940281,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.25,
    }
  }
  locations={
    Array [
      0.2,
      0.3,
    ]
  }
  startPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  style={
    Object {
      "color": "transparent",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      center={null}
      hMargin="0%"
      hPad="0%"
      height="100%"
      justify="center"
      size={null}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "height": 1334,
          "justifyContent": "center",
          "marginHorizontal": 0,
          "marginVertical": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": undefined,
        }
      }
      vMargin="0%"
      vPad="0%"
      width={null}
    >
      <View
        center={null}
        hMargin="0%"
        hPad="2.5%"
        height={null}
        justify="flex-start"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 19,
            "paddingVertical": 73.5,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="5.5%"
        width="100%"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 150,
                "height": 300,
                "width": 300,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/brand/main.jpg",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        center={null}
        hMargin="0%"
        hPad="3%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 22.5,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        >
          <Text>
            Hi
          </Text>
        </View>
      </View>
      <View
        center="end"
        hMargin="0%"
        hPad="0%"
        height={null}
        justify="center"
        size={null}
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "center",
            "marginHorizontal": 0,
            "marginVertical": 0,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "width": 750,
          }
        }
        vMargin="0%"
        vPad="0%"
        width="100%"
      >
        <View
          center={null}
          hMargin="0%"
          hPad="0%"
          height={null}
          justify="center"
          size={null}
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "marginVertical": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "width": undefined,
            }
          }
          vMargin="0%"
          vPad="0%"
          width={null}
        />
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
